### ISSUE TITLE
Adding a `predict_proba` method on SKLearnClassifier

### ISSUE THREAD

**Issue by divakaivan** [keras-team/keras#21547]
Currently the SKLearnClassifier only exposes a `predict` method. At the moment the raw probabilities are used in the predict method like:

```python
# keras/keras/src/wrappers/sklearn_wrapper.py
class SKLBase(BaseEstimator):
    ...
    def predict(self, X):
        """Predict using the model."""
        from sklearn.utils.validation import check_is_fitted

        check_is_fitted(self)
        X = _validate_data(self, X, reset=False)
        raw_output = self.model_.predict(X) # <----- here
        return self._reverse_process_target(raw_output) # <----- transformed back
```

If a user wants to get probabilities out, they need to create a custom class like:

```python
class MyKerasSKLearnClassifier(SKLearnClassifier):
    def predict_proba(self, X):
        if not hasattr(self, 'model_'):
            raise RuntimeError("You must fit the model before calling predict_proba.")
        return self.model_.predict(X) # returns the raw probas
```

Would you consider adding a predict_proba method to the SKLearnClassifier class:

```python
class SKLearnClassifier(ClassifierMixin, SKLBase):
    ...
    def predict_proba(self, X):
        """Predict class probabilities of the input samples X."""
        from sklearn.utils.validation import check_is_fitted

        check_is_fitted(self)
        X = _validate_data(self, X, reset=False)
        return self.model_.predict(X)
```

so that downstream implementations and users can rely on the default class imported from Keras, rather then creating a custom wrapping class. `predict_proba` is the common method in the scikit-learn library for getting the probabilities out of a classifier.

**Comment by divakaivan** [comment]
If ok, I can open a PR :)

**Comment by abheesht17** [comment]
@divakaivan - please go ahead! Do you think we can add a flag to `predict` instead of defining a new method `predict_proba()`? Like so: `predict(..., return_probabilities=False)`

**Comment by divakaivan** [comment]
> [@divakaivan](https://github.com/divakaivan) - please go ahead! Do you think we can add a flag to `predict` instead of defining a new method `predict_proba()`? Like so: `predict(..., return_probabilities=False)`

Thank you. I think it's better to create a `predict_proba` method as some sklearn functionalities and underlying implementations require it. Examples are:

- [calibrating a classifier](https://scikit-learn.org/stable/modules/calibration.html#calibrating-a-classifier)
- [roc_auc_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#roc-auc-score)
- [log_loss](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html#log-loss)
- [roc curve display](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.RocCurveDisplay.html#roccurvedisplay) 

which are commonly used, and if the user wants to use them they have to create a class on top of the SKLearnClassifier again. 

Also, for classifiers in sklearn, it's more common to have `predict_proba` as a separate method.

**Comment by monicadsong** [comment]
Hi @divakaivan --

The Keras team would like to learn more your Keras usage. Please thumbs up this comment if you are open to speaking with us and we'll reach out to LinkedIn. 

Thanks you so much!
Monica, Keras Product Manager
